### PR TITLE
Feat/lw 11756 add drep status provider

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

We want to be able to track DRep statuses from client code.

# Proposed Solution

Add a DRepProvider that can fetch status, example usage:

```
const client = new BlockfrostClient(
    { baseUrl: 'https://cardano-preprod.blockfrost.io', projectId: 'preprodXXXX', apiVersion: 'v0' },
    { rateLimiter: { schedule: (task) => task() } }
);

const dRepProvider = new BlockfrostDRepProvider(client, console);

dRepProvider.getDRepInfo({ id: Cardano.DRepID('drep15cfxz9exyn5rx0807zvxfrvslrjqfchrd4d47kv9e0f46uedqtc') }).then(console.log);

```

Response:
```

{
  active: false,
  activeEpoch: undefined,
  amount: 0n,
  hasScript: true,
  id: 'drep15cfxz9exyn5rx0807zvxfrvslrjqfchrd4d47kv9e0f46uedqtc'
}
```

### CODE REVIEW

File: packages/cardano-services-client/test/DRepProvider/util.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const mockResponses = (request: jest.Mock, responses: [string | RegExp, unknown][]) => {
```
Comment (rhyslbw) [inline packages/cardano-services-client/test/DRepProvider/util.ts:1]: Please hoist https://github.com/input-output-hk/cardano-js-sdk/blob/7198c54ba330233b553501a7d09e0d7ccc4f27ea/packages/cardano-services-client/test/AssetInfoProvider/util.ts#L4 and use that for this provider test. I'll rebase https://github.com/input-output-hk/cardano-js-sdk/pull/1532 to use that change too

File: packages/core/src/Provider/DRepProvider/types.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export interface DRepInfo {
+  id: Cardano.DRepID;
+  amount: Cardano.Lovelace;
+  active: boolean;
+  activeEpoch?: Cardano.EpochNo;
+  hasScript: boolean;
+}
+
+export interface DRepProvider extends Provider {
+  getDRepInfo: (args: GetDRepInfoArgs) => Promise<DRepInfo>;
```
Comment (rhyslbw) [inline packages/core/src/Provider/DRepProvider/types.ts:20]:
```suggestion
```
Precedent it to just have a single endpoint for other providers, with the consumer sending an array of one item to perform individual lookups. e.g https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/core/src/Provider/ChainHistoryProvider/types.ts

File: packages/cardano-services-client/test/DRepProvider/BlockfrostDRepProvider.test.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    const mockedAssetResponse = {
+      active: true,
+      active_epoch: 420,
+      amount: '2000000',
+      drep_id: 'drep15cfxz9exyn5rx0807zvxfrvslrjqfchrd4d47kv9e0f46uedqtc',
+      has_script: true,
+      hex: 'a61261172624e8333ceff098648d90f8e404e2e36d5b5f5985cbd35d'
+    } as Responses['drep'];
+
+    test('has nft metadata', async () => {
```
Comment (rhyslbw) [inline packages/cardano-services-client/test/DRepProvider/BlockfrostDRepProvider.test.ts:31]:
```suggestion
    test('getDRepInfo', async () => {
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: packages/cardano-services-client/test/DRepProvider/util.ts (hunk 1 above)
Comment (AngelCastilloB) [reply packages/cardano-services-client/test/DRepProvider/util.ts]: done here https://github.com/input-output-hk/cardano-js-sdk/pull/1534/commits/574cf3c6ed28010cfa65e10c8391f99340744ac4

File: packages/cardano-services-client/test/DRepProvider/BlockfrostDRepProvider.test.ts (hunk 3 above)
Comment (AngelCastilloB) [reply packages/cardano-services-client/test/DRepProvider/BlockfrostDRepProvider.test.ts]: done here https://github.com/input-output-hk/cardano-js-sdk/pull/1534/commits/574cf3c6ed28010cfa65e10c8391f99340744ac4

File: packages/core/src/Provider/DRepProvider/types.ts (hunk 2 above)
Comment (rhyslbw) [reply packages/core/src/Provider/DRepProvider/types.ts:20]: Ah good point! Precedent is then, in-fact, not clear! OK as-is

**Review by rhyslbw — APPROVED** [review]
Great work @AngelCastilloB ⚡

**Review by mirceahasegan — APPROVED** [review]
Fantastic work @AngelCastilloB 👏 👏 👏

**Pull request merged**